### PR TITLE
Change the algorithm for generated column names to avoid collisions.

### PR DESCRIPTION
**Pull request by BillWhite**
If a user's column name in a subquery happened to be cne of the ones we use for generated
column names we can get wrong answers.  We change the algorithm for generating column
names so that if a nominally named generated name clashes with a user defined name we
add an initial underbar (_).  So, C1 becomes _C1, but only if the user has defined a column
alias C1.

https://issues.voltdb.com/browse/ENG-10497

### CODE REVIEW

File: src/hsqldb19b3/org/hsqldb_voltpatches/ExpressionOp.java [hunk 1]
```diff
@@ -129,7 +129,8 @@
         this.alias = e.alias;
     }
 
-    public String getSQL() {
+    @Override
+        public String getSQL() {
```
Comment (wolffcm) [inline src/hsqldb19b3/org/hsqldb_voltpatches/ExpressionOp.java]: This line should be indented to the same depth as the previous line.

File: src/hsqldb19b3/org/hsqldb_voltpatches/ExpressionOp.java [hunk 2]
```diff
@@ -198,7 +199,8 @@ public String getSQL() {
         return sb.toString();
     }
 
-    protected String describe(Session session, int blanks) {
+    @Override
+        protected String describe(Session session, int blanks) {
```
Comment (wolffcm) [inline src/hsqldb19b3/org/hsqldb_voltpatches/ExpressionOp.java]: Same comment for this line.

Comment (wolffcm) [inline src/hsqldb19b3/org/hsqldb_voltpatches/ExpressionOp.java]: Etc.

**Comment by wolffcm** [comment]
Looks good to me.

File: src/hsqldb19b3/org/hsqldb_voltpatches/FunctionSQL.java [hunk 3]
```diff
@@ -2120,7 +2120,9 @@ public String describe(Session session, int blanks) {
         sb.append(name).append("(");
 
         for (int i = 0; i < nodes.length; i++) {
-            sb.append("[").append(nodes[i].describe(session)).append("]");
+        	if (nodes[i] != null) {
+        		sb.append("[").append(nodes[i].describe(session)).append("]");
+        	}
```
Comment (paulmartel) [inline src/hsqldb19b3/org/hsqldb_voltpatches/FunctionSQL.java]: Non-standard 8-column indents.

File: src/hsqldb19b3/org/hsqldb_voltpatches/QuerySpecification.java [hunk 4]
```diff
@@ -1431,7 +1431,12 @@ void setReferenceableColumns() {
         accessibleColumns = new boolean[indexLimitVisible];
 
         IntValueHashMap aliases = new IntValueHashMap();
-
+        // Bundle up all the user defined aliases here.
+        // We can't import java.util.Set because there is a Set
+        // already imported into this class from Hsql itself.
+        java.util.Set<String> userAliases = new java.util.HashSet<String>();
+        // Bundle up all the generated aliases here.
+        java.util.Map<String, Integer> genAliases = new java.util.HashMap<String, Integer>();
```
Comment (paulmartel) [inline src/hsqldb19b3/org/hsqldb_voltpatches/QuerySpecification.java]: We've lifted the moratorium on changing imports in hsql code -- they'd help readability/conciseness here, as would use of `<>` for the RHS expressions.

**Comment by paulmartel** [comment]
End of review round.

File: src/hsqldb19b3/org/hsqldb_voltpatches/QuerySpecification.java (hunk 4 above)
Comment (BillWhite) [inline src/hsqldb19b3/org/hsqldb_voltpatches/QuerySpecification.java]: HSQL has its own version of Set which conflicts with java.util.Set.  I wanted to use the latter here, but the former is already in use.  So, I need to fully qualify the name.

**Pull request merged**